eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -72,108 +67,93 @@ class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig,
+        router ! createRouteRequest(nodeParams, toSend, maxFee, routeParams, d)
+        stay
       }
-  }
 
-  when(PAYMENT_IN_PROGRESS) {
-    case Event(pf: PaymentFailed, d: PaymentProgress) => handleChildFailure(pf, d) match {
-      case Some(paymentAborted) =>
-        goto(PAYMENT_ABORTED) using paymentAborted
-      case None =>
-        // Get updated local channels (will take into account the child payments that are in-flight).
-        relayer ! GetOutgoingChannels()
-        val failedPayment = d.pending(pf.id)
-        val shouldBlacklist = shouldBlacklistChannel(pf)
-        if (shouldBlacklist) {
-          log.debug(s"ignoring channel ${getFirstHopShortChannelId(failedPayment)} to ${failedPayment.routePrefix.head.nextNodeId}")
-        }
-        val ignoreChannels = if (shouldBlacklist) d.ignoreChannels + getFirstHopShortChannelId(failedPayment) else d.ignoreChannels
-        val remainingAttempts = if (shouldBlacklist && Random.nextDouble() * math.log(d.channelsCount) > 2.0) {
-          // When we have a lot of channels, many of them may end up being a bad route prefix for the destination we're
-          // trying to reach. This is a cheap error that is detected quickly (RouteNotFound), so we don't want to count
-          // it in our payment attempts to avoid failing too fast.
-          // However we don't want to test all of our channels either which would be expensive, so we only probabilistically
-          // count the failure in our payment attempts.
-          // With the log-scale used, here are the probabilities and the corresponding number of retries:
-          //  *   10 channels -> refund 13% of failures -> with 5 initial retries we will actually try 5/(1-0.13) = ~6 times
-          //  *   20 channels -> refund 32% of failures -> with 5 initial retries we will actually try 5/(1-0.32) = ~7 times
-          //  *   50 channels -> refund 50% of failures -> with 5 initial retries we will actually try 5/(1-0.50) = ~10 times
-          //  *  100 channels -> refund 56% of failures -> with 5 initial retries we will actually try 5/(1-0.56) = ~11 times
-          //  * 1000 channels -> refund 70% of failures -> with 5 initial retries we will actually try 5/(1-0.70) = ~17 times
-          // NB: this hack won't be necessary once multi-part is directly handled by the router.
-          d.remainingAttempts + 1
-        } else {
-          d.remainingAttempts
-        }
-        goto(RETRY_WITH_UPDATED_BALANCES) using d.copy(toSend = d.toSend + failedPayment.finalPayload.amount, pending = d.pending - pf.id, failures = d.failures ++ pf.failures, ignoreChannels = ignoreChannels, remainingAttempts = remainingAttempts)
-    }
+    case Event(Status.Failure(t), d: PaymentProgress) =>
+      log.warning("router error: {}", t.getMessage)
+      if (d.ignore.channels.nonEmpty) {
+        // Channels are mostly ignored for temporary reasons, likely because they didn't have enough balance to forward
```

Comment:
```suggestion
        // If no route can be found, we will retry once with the channels that we previously ignored.
        // Channels are mostly ignored for temporary reasons, likely because they didn't have enough balance to forward
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1439/commits/da4f8694edbfc83cfcb3e369079930aac028c748